src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts

```diff
@@ -569,13 +571,18 @@ export class ProgrammingSubmissionService implements IProgrammingSubmissionServi
         const submissions = res.body!;
         const convertedSubmissions: ProgrammingSubmission[] = [];
         for (const submission of submissions) {
-            const latestResult = getLatestSubmissionResult(submission);
-            setLatestSubmissionResult(submission, latestResult);
+            this.convertItemFromServer(submission);
             convertedSubmissions.push({ ...submission });
         }
         return res.clone({ body: convertedSubmissions });
     }
 
+    private convertItemFromServer(programmingSubmission: ProgrammingSubmission): ProgrammingSubmission {
```

Comment:
Just a suggestion but maybe we rename this method so we know it handles specifically latestSubmissions like `convertLatestSubmissionFromServer`? We have similar methods with this name and usually they just convert the given object, here more is done and this should show in the method name i guess

Reply:
yeah, sounds good, I will change this